src/middlewared/middlewared/plugins/smart.py

```diff
@@ -387,10 +387,17 @@ async def manual_test(self, disks):
                         **disk
                     })
                 else:
-                    verrors.add(
-                        f'disks.{index}.identifier',
-                        f'{disk["identifier"]} is not valid. Please provide a valid disk identifier.'
-                    )
+                    disk_info = await self.middleware.call('disk.query', [('identifier', '=', disk['identifier'])], {'get': True})
+                    if disk_info['devname'].startswith('nvme'):
+                        verrors.add(
+                            f'disks.{index}.identifier',
+                            f'{disk["identifier"]} is an nvme disk. We do not support S.M.A.R.T test for nvme disks.'
```

Comment:
Perhaps avoid usage of "We" in validation message.

Reply:
would the following work?
```
S.M.A.R.T tests are not supported for nvme disks for now.
```

Reply:
I see its an experimental feature, we cannot say that its supported on their side.. 🤔